plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1357 @@
+            users_request = {'path': vacm_path, "method": method, 'data': vacm_payload}
+            requests.append(users_request)
+
+        if config.get('view'):
+            views_path = "data/ietf-snmp:snmp/vacm"
+            payload = self.build_create_view_payload(config)
+            views_request = {'path': views_path, 'method': method, 'data': payload}
+            requests.append(views_request)
+
+        return requests
+
+    def build_create_agentaddress_payload(self, config, have, overridden_or_replaced):
+        """ Build the payload for SNMP agentaddress
+        :rtype: A dictionary
+        :returns: The payload for SNMP agentaddress
+        """
+        agentaddress = config.get('agentaddress')
+        have_agentaddress = have.get('agentaddress')
+        have_agentaddress_names = []
+        if have_agentaddress and not overridden_or_replaced:
+            have_agentaddress_names = [agent_entry.get('name') for agent_entry in have_agentaddress]
+        agentaddress_list = []
+        agentaddressdict = {}
+        payload_url = {}
+        for conf in agentaddress:
+            name = ""
+            agentaddress_dict = {}
+            if 'name' in conf:
+                name = conf.get('name')
+            else:
+                name = self.get_agententry(have_agentaddress_names)
+
+            agentaddress_dict['name'] = name
+            agentaddress_dict['udp'] = {'ietf-snmp-ext:interface': conf.get('interface'), 'ip': conf.get('ip'), 'port': conf.get('port')}
+            agentaddress_list.append(agentaddress_dict)
+
+        agentaddressdict['listen'] = agentaddress_list
+        payload_url['ietf-snmp:engine'] = agentaddressdict
+        return payload_url
+
+    def build_create_community_payload(self, config, have):
+        """ Build the payload for SNMP community
+        :rtype: A dictionary
+        :returns: The list of community requests
+        """
+        community = config.get('community')
+        community_list = []
+        payload_url = {}
+        community_requests = []
+        community_path = "data/ietf-snmp:snmp/community"
+
+        for conf in community:
+            community_dict = {}
+            community_dict['index'] = conf.get('name')
+
+            if have and len(have) > 0 and have.get('group'):
+                for group in have.get(['group']):
+                    matched_group = next((each_group for each_group in group.get('member') if each_group['security-name'] == conf.get('name')), None)
+                    if matched_group:
+                        group_name = group.get('name')
+                        group_url = "data/ietf-snmp:snmp/vacm/group={0}/member={1}".format(group_name, matched_group.get('security-name'))
+                        group_request = {"path": group_url, "method": DELETE}
+                        community_requests.append(group_request)
+
+            community_dict['security-name'] = conf.get('name')
```

Comment:
The "security-name" for the community should be the name of the group to which it is being added.

```suggestion
            community_dict['security-name'] = conf.get('group')
```

Also, if the instruction that was previously near the top of the "for conf in community" block is restored:

`group_name = conf.get('group')`

it would be possible to use "group_name" here and in the previously commented block at line 472 to avoid fetching it twice (with the "conf.get('group') instructions currently shown in both places.